### PR TITLE
Fix up some unnecessary uses of `unsafe`.

### PR DESCRIPTION
**Pull request by eefriedman**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7523)

### CODE REVIEW

File: components/gfx/font.rs [hunk 1]
```diff
@@ -56,12 +55,11 @@ pub trait FontTableTagConversions {
...
-            let pointer = mem::transmute::<&u32, *const u8>(self);
-            let mut bytes = slice::from_raw_parts(pointer, 4).to_vec();
-            bytes.reverse();
-            String::from_utf8_unchecked(bytes)
-        }
+        let bytes = [(self >> 24) as u8,
+                     (self >> 16) as u8,
+                     (self >>  8) as u8,
+                     (self >>  0) as u8];
+        str::from_utf8(&bytes).unwrap().to_owned()
```
Comment (SimonSapin) [inline components/gfx/font.rs]: This changes the behavior of big-endian platforms, but I don’t know which behavior is correct.

Comment (SimonSapin) [inline components/gfx/font.rs]: Looks like this is only used for a `debug!` line, so maybe it doesn’t matter that much.

Comment (eefriedman) [inline components/gfx/font.rs]: This is the same way harfbuzz handles tags.  See http://mxr.mozilla.org/mozilla-central/source/gfx/harfbuzz/src/hb-common.h#94 .

**Pull request merged**